mitmproxy-rs/src/udp_client.rs

```diff
@@ -102,7 +102,7 @@ pub struct UdpClientTask {
 
 impl UdpClientTask {
     pub async fn run(mut self) -> Result<()> {
-        let mut udp_buf = [0; MAX_PACKET_SIZE];
+        let mut udp_buf = Box::new([0; MAX_PACKET_SIZE]);
```

Comment:
`MAX_PACKET_SIZE` in the stack explains why stack frames are pretty big. Moving this to the heap sounds like a good idea - but the syntax you're using here might not do what you want it to. It's a common footgun with allocating arrays on the heap:

https://stackoverflow.com/questions/30242770/how-to-allocate-arrays-on-the-heap-in-rust-1-0

Reply:
Super great spot, fixed! 🍰